apps/admin/frontend/src/screens/tally_report_screen.test.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { electionMinimalExhaustiveSampleDefinition } from '@votingworks/fixtures';
+import {
+  advanceTimersAndPromises,
+  fakeKiosk,
+  fakePrinterInfo,
+} from '@votingworks/test-utils';
+import { fakeLogger, Logger } from '@votingworks/logging';
+import { screen, waitFor, within } from '@testing-library/react';
+
+import { Admin } from '@votingworks/api';
+import { renderInAppContext } from '../../test/render_in_app_context';
+import {
+  ApiMock,
+  createApiMock,
+  createMockApiClient,
+  MockApiClient,
+} from '../../test/helpers/api_mock';
+import { TallyReportScreen } from './tally_report_screen';
+import { routerPaths } from '../router_paths';
+
+jest.mock('../components/hand_marked_paper_ballot');
+
+let mockKiosk: jest.Mocked<KioskBrowser.Kiosk>;
+let logger: Logger;
+let apiClient: MockApiClient;
+let apiMock: ApiMock;
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  mockKiosk = fakeKiosk();
+  mockKiosk.getPrinterInfo.mockResolvedValue([
+    fakePrinterInfo({ connected: true, name: 'VxPrinter' }),
+  ]);
+  window.kiosk = mockKiosk;
+  logger = fakeLogger();
+  apiClient = createMockApiClient();
+  apiMock = createApiMock(apiClient);
+});
+
+afterAll(() => {
+  delete window.kiosk;
+  apiClient.assertComplete();
+});
+
+test('mark official results button disabled when no cast vote record files', async () => {
+  apiMock.expectGetWriteInSummaryAdjudicated([]);
+  apiMock.expectGetCastVoteRecordFileMode(Admin.CvrFileMode.Unlocked);
+  renderInAppContext(<TallyReportScreen />, {
+    electionDefinition: electionMinimalExhaustiveSampleDefinition,
+    logger,
+    apiClient,
+    route: routerPaths.tallyFullReport,
+  });
+
+  await waitFor(() => {
+    apiMock.apiClient.assertComplete();
+  });
+  await advanceTimersAndPromises(); // wait for UI to settle
+  expect(
+    screen.getByRole('button', {
+      name: 'Mark Tally Results as Official',
+    })
+  ).toBeDisabled();
+});
+
+test('when already official results', async () => {
+  apiMock.expectGetWriteInSummaryAdjudicated([]);
+  apiMock.expectGetCastVoteRecordFileMode(Admin.CvrFileMode.Official);
+  renderInAppContext(<TallyReportScreen />, {
+    electionDefinition: electionMinimalExhaustiveSampleDefinition,
+    isOfficialResults: true,
+    logger,
+    apiClient,
+    route: routerPaths.tallyFullReport,
+  });
+
+  await waitFor(() => {
+    apiMock.apiClient.assertComplete();
+  });
```

Comment:
In this case, we have a button that is disabled and is only enabled once the information from the backend is loaded. If we check that it's disabled right out the gate, it will always be true, and not actually do the test properly.

Reply:
Oh, I see. That makes sense. If there's no other visual indicator on the screen, then I can't think of any other way. Maybe add an explanatory comment?

I do worry a bit that this test is kind of brittle. What's the guarantee that the UI will "settle" after `advanceTimersAndPromises()`?